src/main/java/pl/agawesolowska/ticketbookingapp/config/RestConfiguration.java

```diff
@@ -0,0 +1,98 @@
+package pl.agawesolowska.ticketbookingapp.config;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import javax.persistence.EntityManager;
+import javax.persistence.metamodel.EntityType;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.rest.core.config.RepositoryRestConfiguration;
+import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurer;
+import org.springframework.http.HttpMethod;
+
+import pl.agawesolowska.ticketbookingapp.model.entity.Booking;
+import pl.agawesolowska.ticketbookingapp.model.entity.Cinema;
+import pl.agawesolowska.ticketbookingapp.model.entity.Customer;
+import pl.agawesolowska.ticketbookingapp.model.entity.Room;
+import pl.agawesolowska.ticketbookingapp.model.entity.Row;
+import pl.agawesolowska.ticketbookingapp.model.entity.Screening;
+import pl.agawesolowska.ticketbookingapp.model.entity.Seat;
+
+/**
+ * @author Aga Wesołowska
+ *
+ */
+@Configuration
+public class RestConfiguration implements RepositoryRestConfigurer {
+
+	private final EntityManager entityManager;
+
+	@Autowired
+	public RestConfiguration(EntityManager entityManager) {
+		this.entityManager = entityManager;
+	}
+
+	@Override
+	public void configureRepositoryRestConfiguration(RepositoryRestConfiguration config) {
```

Comment:
jak jest idea tej konfiguracji? nie da się tego prościej?

Reply:
Zaczerpnęłam te konfigurację z kursu na Udemy, który robiłam: apka e-commerce z wykorzystaniem Spring Data REST. Chciałam zablokować możliwość edycji/kasowania danych w bazie przy wykorzystaniu metod POST, PUT i DELETE, czyli ustawić konfigurację 'read-only' - w założeniu projektu użytkownik rezerwuje bilety na seans, nie ma możliwości np. dodania do bazy kolejnego seansu. Plus zależało mi na tym, żeby wystawić wartości id, których używam w kolejnych requestach.
Decydując się na Spring Data REST miałam świadomość, że dane z bazy są wystawione na zewnątrz, i:
1) nie muszę powielać kontrolerów dla poszczególnych entities,
2) ale, bez blokowania tych metod HTTP można wykonać np. taki niepożądany request:
`curl --location --request DELETE 'http://localhost:8080/api/v1/screenings/1'`
Można to prościej ustawić dzięki adnotacji @Transactional(readOnly = true) nad każdym z Repository, unikając powielania kodu. Zdecydowałam się na brzydszy, ale pewniejszy dla mnie sposób.